src/components/main/ImportH5P.tsx

```diff
@@ -1,72 +1,97 @@
-  onComplete,
+  onClose,
   previousItemId,
 }: {
-  onComplete?: () => void;
+  onClose?: () => void;
   previousItemId?: DiscriminatedItem['id'];
 }): JSX.Element => {
   const { itemId } = useParams();
   const { mutateAsync: importH5P, isPending: isLoading } =
     mutations.useImportH5P();
   const { update, close: closeNotification } = useUploadWithProgress();
   const { t: translateBuilder } = useBuilderTranslation();
+  const { t: translateCommon } = useCommonTranslation();
 
   return (
-    <Box overflow="auto">
-      <Typography variant="body1" paragraph>
-        {translateBuilder(BUILDER.IMPORT_H5P_INFORMATIONS)}
-      </Typography>
-      <Typography variant="body1" paragraph>
-        {translateBuilder(BUILDER.IMPORT_H5P_WARNING)}
-      </Typography>
-      <Typography variant="body1" paragraph>
-        {translateBuilder(BUILDER.IMPORT_H5P_LIMITATIONS_TEXT, {
-          maxSize: formatFileSize(MAX_ZIP_FILE_SIZE),
-        })}
-      </Typography>
-      <UploadFileButton
-        isLoading={isLoading}
-        loadingText={translateBuilder(BUILDER.UPLOADING)}
-        onChange={(e) => {
-          if (e.target.files?.length) {
-            importH5P({
-              onUploadProgress: update,
-              id: itemId,
-              previousItemId,
-              file: e.target.files[0],
-            })
-              .then(() => {
-                closeNotification();
-                onComplete?.();
-              })
-              .catch((error) => {
-                closeNotification(error);
-              });
-          }
-        }}
-        accept=".h5p"
-        id={H5P_DASHBOARD_UPLOADER_ID}
-        text={translateBuilder(BUILDER.UPLOAD_H5P_BUTTON)}
-      />
-    </Box>
+    <>
+      <DialogTitle>{translateBuilder(BUILDER.IMPORT_H5P_TITLE)}</DialogTitle>
+      <DialogContent>
+        <Stack gap={1}>
+          <Typography variant="body1">
+            {translateBuilder(BUILDER.IMPORT_H5P_INFORMATIONS)}
+          </Typography>
+          <Typography variant="body1">
+            {translateBuilder(BUILDER.IMPORT_H5P_WARNING)}
+          </Typography>
+          <Typography variant="body1">
+            {translateBuilder(BUILDER.IMPORT_H5P_LIMITATIONS_TEXT, {
+              maxSize: formatFileSize(MAX_ZIP_FILE_SIZE),
+            })}
+          </Typography>
+          <UploadFileButton
+            isLoading={isLoading}
+            loadingText={translateBuilder(BUILDER.UPLOADING)}
+            onChange={(e) => {
+              if (e.target.files?.length) {
+                importH5P({
+                  onUploadProgress: update,
+                  id: itemId,
+                  previousItemId,
+                  file: e.target.files[0],
+                })
+                  .then(() => {
+                    closeNotification();
+                    onClose?.();
+                  })
+                  .catch((error) => {
+                    closeNotification(error);
+                  });
```

Comment:
```suggestion
                try {
                  await importH5P({
                    onUploadProgress: update,
                    id: itemId,
                    previousItemId,
                    file: e.target.files[0],
                  });
                  
                  closeNotification();
                  onClose?.();
                } catch (error) {
                    closeNotification(error);
                };
```

We could also handle this logic in a function?